web/html/src/components/large-text-attachment.tsx

```diff
@@ -0,0 +1,251 @@
+import * as React from "react";
+
+import { Button, LinkButton } from "components/buttons";
+import { DangerDialog } from "components/dialog/DangerDialog";
+import { Dialog } from "components/dialog/Dialog";
+import { Form, Radio, Text, TextArea } from "components/input";
+import { showErrorToastr } from "components/toastr";
+
+enum EditMethod {
+  Upload = "upload",
+  Paste = "paste",
+}
+
+type EditFormModel = {
+  method: EditMethod;
+  uploadedValue?: string;
+  pastedValue?: string;
+};
+
+type Props = {
+  /** The text data, can be null */
+  value: string | null;
+  /** The filename to be suggested to the user for the download */
+  filename: string;
+  /** Hide the text message before the action buttons */
+  hideMessage: boolean;
+  /** The message to show when the data is present */
+  presentMessage?: string;
+  /** The message to show when the data is null */
+  absentMessage?: string;
+  /** Defines if the component allows editing the data */
+  editable: boolean;
+  /** The title of the editing dialog, used for both modification and creation */
+  editDialogTitle: string;
+  /** A custom message displayed in the editing dialog, before the form */
+  editMessage?: string;
+  /** Message displayed on the delete confirmation dialog */
+  confirmDeleteMessage: string;
+  /** Callback to invoked when the user confirms on the edit dialog */
+  onEdit: (value: string) => Promise<any>;
+  /** Callback to invoked when the user confirms on the delete dialog */
+  onDelete: () => Promise<any>;
+};
+
+type State = {
+  showDeleteDialog: boolean;
+  showEditDialog: boolean;
+  editFormModel: EditFormModel;
+  editFormValidated: boolean;
+};
+
+/**
+ * Component to handle large text data. It allows the user to download, edit and delete the text data.
+ */
+export class LargeTextAttachment extends React.Component<Props, State> {
+  private static readonly INITAL_MODEL: EditFormModel = { method: EditMethod.Upload };
+
+  static defaultProps: Partial<Props> = {
+    filename: "attachment.txt",
+    editable: false,
+    hideMessage: false,
+    presentMessage: t("Data is present"),
+    absentMessage: t("No data available."),
```

Comment:
This is a very small nitpick, but I would recommend considering if we can use consistent language here. E.g. data is present and not present or data is available and not available. Currently it's unclear from the user's perspective whether the distinction between present/available is important or not.

Reply:
Fixed